packages/authentication/docs/controller-functions.md

```diff
@@ -0,0 +1,106 @@
+## Endpoint definitions
+
+The following decorated controller functions demos the endpoints described at
+the beginning of markdown file
+[authentication-system](./authentication-system.md).
+
+Please note how they are decorated with `@authenticate()`, the syntax is:
+`@authenticate(<strategy_name>, {action: <action_name>, session: <enabled_or_not>})`
+
+- /login
+
+```ts
+class LoginController {
+  @post('/login', APISpec)
+  login() {
+    // static route
+  }
+}
+```
+
+- /loginWithLocal
+
+```ts
+
+const RESPONSE_SPEC_FOR_JWT_LOGIN = {
+    responses: {
+      '200': {
+        description: 'Token',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'object',
+              properties: {
+                token: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+};
+
+class LoginController{
+  constructor(
+    @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+    @inject(AuthenticationBindings.SERVICES.JWT_TOKEN) JWTtokenService: TokenService,
+  ) {}
+
+  // Describe the response using OpenAPI spec
+  @post('/loginOAI/local', RESPONSE_SPEC_FOR_JWT_LOGIN)
+  @authenticate('basicAuth')
```

Comment:
Do we allow a method to be authenticated by one of the strategies?
```ts
@authenticate(['basicAuth', 'localAuth']) // We allow either basic or local auth
```

Reply:
I thought "local" and "basic" auth are same, which is wrong. So the endpoint here meant to be decorated with only one strategy.

Reply:
We leave it for future decisions. It would be nice to support one of the many auths on the same method.

Reply:
> We leave it for future decisions. It would be nice to support one of the many auths on the same method.

My understanding is:
- For "login" endpoints:
  - the number of endpoints will be small, e.g. one system supports 5 login ways
  - usually there is only one strategy applied per endpoint
- For "verify" endpoints:
  - the number of endpoints will be large
  - it makes sense to specify a default auth strategy for them instead of repeating the decoration thousands of times, OR, to be more practical, resolve the auth strategy based on the login approach.

There is an issue discussing about resolving the strategy after login:
Shall we continue the discussion in https://github.com/strongloop/loopback-next/issues/2139?